### PR TITLE
Trigger

### LINKED ISSUE #80: Trigger
Like we have [Trigger](https://kestra.io/plugins/plugin-gcp/triggers/bigquery/io.kestra.plugin.gcp.bigquery.Trigger.html) on BigQuery for example, we need the same for all JDBC implementation

### PR DESCRIPTION
**Pull request by loicmathieu**
Closes #80

### CODE REVIEW

**Comment by loicmathieu** [comment]
@tchiotludo, the rockset test is not working I don't know why, but the PR is ready to me reviewed.

There is a lot of changes as I had to refactor things to share the conf between Query and Trigger, it should be easier to review commit by commit.

File: plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Query.java [hunk 1]
```diff
@@ -77,24 +77,30 @@ public class Query extends AbstractJdbcQuery implements RunnableTask<Query.Outpu
     @PluginProperty(dynamic = false)
     protected List<String> outputFiles;
 
+    private Path databaseFile;
```
Comment (tchiotludo) [inline plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Query.java:80]: You should add a `@Getter(AccessLevel.NONE)` and a transient, if not the properties will be public

File: plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Trigger.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                "  - id: watch",
+                "    type: io.kestra.plugin.jdbc.duckdb.Trigger",
+                "    interval: \"PT5M\"",
+                "    sql: \"SELECT * FROM my_table\""
+            }
+        )
+    }
+)
+public class Trigger extends AbstractJdbcTrigger {
+    private Path databaseFile;
```
Comment (tchiotludo) [inline plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Trigger.java:65]: Same for transient & getter

File: plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Query.java [hunk 3]
```diff
@@ -50,16 +51,19 @@
 public class Query extends AbstractJdbcQuery implements RunnableTask<AbstractJdbcQuery.Output>, AutoCommitInterface {
     protected final Boolean autoCommit = true;
 
+    @PluginProperty(dynamic = true)
```
Comment (tchiotludo) [inline plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Query.java:54]: Make a interface to have a common documenation

File: plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Trigger.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                "    type: io.kestra.plugin.jdbc.rockset.Trigger",
+                "    interval: \"PT5M\"",
+                "    sql: \"SELECT * FROM my_table\""
+            }
+        )
+    }
+)
+public class Trigger extends AbstractJdbcTrigger {
+
+    @PluginProperty(dynamic = true)
```
Comment (tchiotludo) [inline plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Trigger.java:57]: Make a interface to have a common documenation

**Review by tchiotludo — CHANGES_REQUESTED** [review]

File: plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Query.java (hunk 1 above)
Comment (loicmathieu) [reply plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Query.java:80]: This was existing in the Query task, I'll update both places.

Comment (loicmathieu) [reply plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Query.java]: done

File: plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Trigger.java (hunk 2 above)
Comment (loicmathieu) [reply plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Trigger.java]: done

File: plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Trigger.java (hunk 4 above)
Comment (loicmathieu) [reply plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Trigger.java]: done

File: plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Query.java (hunk 3 above)
Comment (loicmathieu) [reply plugin-jdbc-rockset/src/main/java/io/kestra/plugin/jdbc/rockset/Query.java]: done

**Pull request merged**